Medicare/src/DatabaseAccessLayer/DAO.java

```diff
@@ -71,420 +74,238 @@ public ResultSet getRs() {
+        } catch (Exception e) {
+            // this.txtMsg.setText("Failed to connect; Please viewStack Trace");
             connectionFlag = false;
-               StringBuilder sb = new StringBuilder(e.toString());
-                for (StackTraceElement ste : e.getStackTrace()) {
-                    sb.append("\n\tat ");
-                    sb.append(ste);
-                }
-                String trace = sb.toString();
-            JOptionPane.showMessageDialog(null, trace,"Message", JOptionPane.INFORMATION_MESSAGE);
+            StringBuilder sb = new StringBuilder(e.toString());
+            for (StackTraceElement ste : e.getStackTrace()) {
+                sb.append("\n\tat ");
+                sb.append(ste);
+            }
+            String trace = sb.toString();
+            JOptionPane.showMessageDialog(null, trace, "Message", JOptionPane.INFORMATION_MESSAGE);
         }
 
-     }
-    
-
-
-     public String[] getHealthCareTypes( ) {
-         String [] healthCarType;
-         try
-         {
-            stmt = con.createStatement();   
-            String sql = "select * from HEALTHCARE";
-            rs= stmt.executeQuery(sql);
-           
-            ArrayList<String> result = new ArrayList<String>();
-            
-             while(rs.next())
-            {
-                result.add(rs.getString(2));
-            }
+    }
 
-            
-            System.out.println(result.toString());
-             return (String[]) result.toArray(new String[result.size()]);
-                
-         }catch(SQLException e)
-         {
-             e.printStackTrace();
-         }
-        return null;
-     }
-     
-     public String[] getEmergecyTypes( ) {
-        
-         try
-         {
-            stmt = con.createStatement();   
-            String sql = "select * from emer_svcs";
-            rs= stmt.executeQuery(sql);
-           
-            ArrayList<String> result = new ArrayList<String>();
-            
-             while(rs.next())
-            {
-                result.add(rs.getString(2));
-            }
+    public ResultSet fetchDataSource() {
 
-            
-            System.out.println(result.toString());
-             return (String[]) result.toArray(new String[result.size()]);
-                
-         }catch(SQLException e)
-         {
-             e.printStackTrace();
-         }
-        return null;
-     }
-
-      public String[] getInPatientTypes( ) {
-        
-         try
-         {
-            stmt = con.createStatement();   
-            String sql = "select * from indoor_patient";
-            rs= stmt.executeQuery(sql);
-           
-            ArrayList<String> result = new ArrayList<String>();
-            
-             while(rs.next())
-            {
-                result.add(rs.getString(2));
-            }
+        try {
+            stmt = con.createStatement();
+            String  sql = "call dayEndReport()";

```

Comment:
dont hardcode the procedure names here. declare them as constants in some constants file.

call dayEndReport()